app/presenters/results_presenter.rb

```diff
@@ -0,0 +1,70 @@
+module Presenters
+  class ResultsPresenter
+    def initialize(results, pagination, request_url)
+      @results = results
+      @pagination = pagination
+      @request_url = request_url.to_s
+    end
+
+    def present
+      {
+        total: results.total,
+        pages: pagination_pages,
+        current_page: current_page,
+        links: links,
+        results: results.call
+      }
+    end
+
+  private
+
+    attr_reader :results, :pagination, :request_url
+
+    def current_page
+      pagination.page.to_i
+    end
+
+    def pagination_pages
+      pagination.pages(results.total)
+    end
+
+    def links
+      [previous_link, next_link, self_link].compact
+    end
+
+    def previous_link
+      return unless previous_page?
+      {
+        href: page_href(:-),
+        rel: "previous"
+      }
+    end
+
+    def next_link
+      return unless next_page?
+      {
+        href: page_href(:+),
+        rel: "next"
+      }
+    end
+
+    def self_link
+      {
+        href: page_href(:+, 0),
+        rel: "self"
+      }
+    end
+
+    def page_href(operator, offset = 1)
+      request_url.gsub(/&page=(\d*)/, "") << "&page=#{current_page.send(operator, offset)}"
```

Comment:
This is a bit magic. Can we not just use a negative offset to get the previous page?